### PR TITLE
Pin versions to tag, commit or release

### PR DESCRIPTION
**Pull request by jovial**
Kind of obscures which openstack release you are using... thoughts?

### CODE REVIEW

File: etc/kayobe/bifrost.yml [hunk 1]
```diff
@@ -5,10 +5,10 @@
 # Bifrost installation.
 
 # URL of Bifrost source code repository.
-#kolla_bifrost_source_url:
+kolla_bifrost_source_url: https://github.com/stackhpc/bifrost.git
 
 # Version (branch, tag, etc.) of Bifrost source code repository.
-#kolla_bifrost_source_version:
+kolla_bifrost_source_version: refs/tags/stackhpc/3.0.0.1
```
Comment (markgoddard) [inline etc/kayobe/bifrost.yml]: The latest pike release of bifrost is 4.0.1. Although in fact we're running ocata - we never braved an upgrade of the seed. I guess we ought to keep the config updated for pike though.

File: etc/kayobe/kolla.yml [hunk 2]
```diff
@@ -103,30 +103,30 @@ kolla_build_blocks:
         && rm -rf /fluentd-monasca
   neutron_server_footer: |
     # We need a patch neutron to force the provision network MTU to 1500.
-    RUN pip install -U --no-deps git+https://github.com/stackhpc/neutron.git@stackhpc/pike
+    RUN pip install -U --no-deps git+https://github.com/stackhpc/neutron.git@stackhpc/11.0.0.0b1.1
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: I don't think we need neutron fork any more - the MTU issue we hacked around is now gone. We can remove this line.

File: etc/kayobe/kolla.yml [hunk 3]
```diff
@@ -103,30 +103,30 @@ kolla_build_blocks:
...
+    RUN pip install git+https://github.com/stackhpc/networking-generic-switch@stackhpc/0.4.1.1
     # Install a locally modified version of networking-mlnx with support for
     # disabling the SDN controller sync mechanism in the mlnx_sdn_assist
     # driver. This allows us to bind to flat Infiniband networks.
-    RUN pip install git+https://github.com/stackhpc/networking-mlnx@cauterise-sdn-client
+    RUN pip install git+https://github.com/stackhpc/networking-mlnx@stackhpc/11.0.0.1
   ironic_base_footer: |
     # Pull in Moshe's fix for IB ports to ensure they don't need
     # local-link-connection information:
     # https://review.openstack.org/#/c/548396. This can be removed once the
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: This patch is included in pike release 9.1.3. RDO provides 9.1.5, so we can now remove the ironic_base_footer section.

File: etc/kayobe/kolla.yml [hunk 4]
```diff
@@ -135,10 +135,10 @@ kolla_build_blocks:
         && pip install -U git+https://github.com/stackhpc/monasca-log-api@stackhpc/log-query-api
   sahara_api_footer: |
     # Bring in hack to the vanilla plugin
-    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/pike
+    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/7.0.0.0b2.1
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: Git describe on stable/pike says `7.0.2-12-g7402129`

File: etc/kayobe/kolla.yml [hunk 5]
```diff
@@ -135,10 +135,10 @@ kolla_build_blocks:
...
   sahara_engine_footer: |
     # Bring in hack to the vanilla plugin
-    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/pike
+    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/7.0.0.0b2.1
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: As above.

**Review by markgoddard — CHANGES_REQUESTED** [review]
We could also pin IPA, via `ipa_build_source_version`.

File: etc/kayobe/kolla.yml (hunk 4 above)
Comment (markgoddard) [reply etc/kayobe/kolla.yml]: The b releases are betas.

File: etc/kayobe/kolla.yml [hunk 6]
```diff
@@ -135,10 +127,10 @@ kolla_build_blocks:
         && pip install -U git+https://github.com/stackhpc/monasca-log-api@stackhpc/log-query-api
   sahara_api_footer: |
     # Bring in hack to the vanilla plugin
-    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/7.0.0.0b2.1
+    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/7.0.2.1
   sahara_engine_footer: |
     # Bring in hack to the vanilla plugin
-    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/7.0.0.0b2.1
+    RUN pip install -U --no-deps git+https://github.com/stackhpc/sahara.git@stackhpc/stackhpc/7.0.2.1
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: I know you love StackHPC, but one is sufficient here

Comment (jovial) [reply etc/kayobe/kolla.yml]: Nice spot - damnit!

**Review by markgoddard — APPROVED** [review]

**Pull request merged**